components/operations-controller/main.go

```diff
@@ -132,16 +123,27 @@ func fatalOnError(err error) {
 	}
 }
 
-func defaultOAuthClientProviderFunc(ctx context.Context, client http.Client, oauthCreds *director_graphql.OAuthCredentialData) *http.Client {
-	conf := &clientcredentials.Config{
-		ClientID:     oauthCreds.ClientID,
-		ClientSecret: oauthCreds.ClientSecret,
-		TokenURL:     oauthCreds.URL,
-		AuthStyle:    oauth2.AuthStyleInParams,
+func prepareHttpClient(cfg *httputil.Config, graphqlCfg *graphql.Config) (*http.Client, error) {
```

Comment:
passing the graphql target url here is a bit strange, also the unsigned provider uses this method, others return nil, so this probably means that the abstraction of target url does not fit well here. Basically now the secure transport has to understand that not all providers provide a target url and needs to provide its own value then.

Ideally the directorURL passed to the director client should be reused and targeturl() removed from interface - you should pass it to PrepareGqlClient and provide it in 	graphClient := graphql.NewClient("", graphql.WithHTTPClient(httpClient)) where currently an empty string is passed as url